src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp

```diff
@@ -336,11 +336,13 @@ void Transformations::PreLpt(const std::vector<ov::element::Type>& defaultPrecis
         if (!hasHardwareSupport(ov::element::bf16))
             map.insert({ov::element::bf16, ov::element::f32});
 #if defined(OPENVINO_ARCH_ARM) || defined(OPENVINO_ARCH_ARM64)
-        if (inferencePrecision != ov::element::f16) {
-                map.insert({ov::element::f16, ov::element::f32});
+        if (!one_of(inferencePrecision, element::f16, element::undefined)) {
+            map.insert({element::f16, element::f32});
         }
 #else
-        map.insert({ov::element::f16, ov::element::f32});
+        if (inferencePrecision != element::undefined) {
+            map.insert({element::f16, element::f32});
+        }
```

Comment:
fq nodes converted from f16 to f32 if f16 precision is selected (356-365 lines in this file).
We do it to avoid performance degradation. Should we skip this convert if undefined precision is selected?

Reply:
The `undefined` type can be set intentionally only for Performance mode and should not affect existing perf scenarios.
This type may affect Accuracy mode, since it's used by default there.

Reply:
Those code is executed just if `inferencePrecision == ov::element::f16`, so no need to change it.